components/post_view/message_attachments/action_menu/index.ts

```diff
@@ -28,14 +27,14 @@ function mapStateToProps(state: GlobalState, ownProps: OwnProps) {
     };
 }
 
-function mapDispatchToProps(dispatch: Dispatch<GenericAction>) {
-    return {
-        actions: bindActionCreators({
-            selectAttachmentMenuAction,
-            autocompleteChannels,
-            autocompleteUsers,
-        }, dispatch),
-    };
-}
+const mapDispatchToProps = {
+    selectAttachmentMenuAction,
+    autocompleteChannels,
+    autocompleteUsers,
+};
+
+const connector = connect(mapStateToProps, mapDispatchToProps);
+
+export type PropsFromRedux = ConnectedProps<typeof connector>;
```

Comment:
This was important as we just did the migration of 'autocompleteChannels' to ts. And the ts compiler was failing here. Converting to direclty PropsFromRedux solves the issue

Reply:
If it works, then great, but it feels "out of the ordinary". 0/5 if we should try to follow the current pattern, or even consider make out of this a new pattern.

Reply:
Yes we discussed this pattern in last web guild.

Reply:
even better would be just useDispatch. but only applied to functional components